docker_test.go

```diff
@@ -1033,8 +1034,93 @@ func Test_BuildContainerFromDockerfile(t *testing.T) {
 		WaitingFor:   wait.ForLog("Ready to accept connections"),
 	}
 
-	t.Log("creating generic container request from container request")
+	redisC, err := prepareRedisImage(ctx, req, t)
+	require.NoError(t, err)
+	terminateContainerOnEnd(t, ctx, redisC)
+
+	checkSuccessfulRedisImage(ctx, redisC, t)
+
+}
+
+func Test_BuildContainerFromDockerfileWithAuthConfig_ShouldSucceedWithAuthConfigs(t *testing.T) {
+	prepareLocalRegistryWithAuth(t)
+	defer func() {
+		ctx := context.Background()
+		testcontainersClient, err := client.NewClientWithOpts(client.WithVersion(daemonMaxVersion))
+		if err != nil {
+			t.Log("could not create client to cleanup registry: ", err)
+		}
+
+		_, err = testcontainersClient.ImageRemove(ctx, "localhost:5000/redis:5.0-alpine", types.ImageRemoveOptions{
+			Force:         true,
+			PruneChildren: true,
+		})
+		if err != nil {
+			t.Log("could not remove image: ", err)
+		}
+
+	}()
+
+	t.Log("getting context")
+	ctx := context.Background()
+	t.Log("got context, creating container request")
+	req := ContainerRequest{
+		FromDockerfile: FromDockerfile{
+			Context:    "./testresources",
+			Dockerfile: "auth.Dockerfile",
+			AuthConfigs: map[string]types.AuthConfig{
+				"localhost:5000": {
+					Username: "testuser",
+					Password: "testpassword",
+				},
+			},
+		},
+		ExposedPorts: []string{"6379/tcp"},
+		WaitingFor:   wait.ForLog("Ready to accept connections"),
+	}
+
+	redisC, err := prepareRedisImage(ctx, req, t)
+	require.NoError(t, err)
+	terminateContainerOnEnd(t, ctx, redisC)
 
+	checkSuccessfulRedisImage(ctx, redisC, t)
+}
+
+func Test_BuildContainerFromDockerfileWithAuthConfig_ShouldFailWithoutAuthConfigs(t *testing.T) {
+	prepareLocalRegistryWithAuth(t)
+
+	t.Log("getting context")
+	ctx := context.Background()
+	t.Log("got context, creating container request")
+	req := ContainerRequest{
+		FromDockerfile: FromDockerfile{
+			Context:    "./testresources",
+			Dockerfile: "auth.Dockerfile",
+		},
+		ExposedPorts: []string{"6379/tcp"},
+		WaitingFor:   wait.ForLog("Ready to accept connections"),
+	}
+
+	redisC, err := prepareRedisImage(ctx, req, t)
+	require.Error(t, err)
+	terminateContainerOnEnd(t, ctx, redisC)
+}
+
+func prepareLocalRegistryWithAuth(t *testing.T) {
+	compose, err := NewDockerCompose("./testresources/docker-compose-auth.yml")
```

Comment:
I think you can start a generic container here instead of the compose one 🤔

Reply:
Sure but, do you see an issue with being a docker-compose instead?

Reply:
For the pure writing and maintenance of the tests, they would be more clear as a GenericContainer, as the definition of the container would be there in the test method, instead of in a compose file that we have to open elsewhere.

I'd advocate for writing compose tests for the compose code.

Reply:
Done. I've replaced the docker-compose with a docker container in preparation to the actual test.